kpatch-build/lookup.c

```diff
@@ -65,6 +66,58 @@ struct lookup_table {
 #define for_each_exp_symbol(ndx, iter, table) \
 	for (ndx = 0, iter = table->exp_syms; ndx < table->exp_nr; ndx++, iter++)
 
+static void find_local_syms(struct lookup_table *table, char *hint,
+			    struct sym_compare_type *locals)
+{
+	struct object_symbol *sym, *file_sym;
+	int i, in_file = 0;
+	struct sym_compare_type *local_index;
+
+	for_each_obj_symbol(i, sym, table) {
+		if (sym->type == STT_FILE) {
+			if (in_file) {
```

Comment:
I found a bug here, when trying to patch fs/proc/cmdline.c. It gives the "find_local_syms found_dup" error, because there's another cmdline.c which has no local symbols. It treats an empty file as a match.

It should instead make sure that local_index->name is zero before checking the error and/or assigning table->local_syms.  So I think this line should be `if (in_file && !local_index->name) {` and a similar check below at line 108.